### PR TITLE
Add dnf module

### PR DESCRIPTION
**Pull request by tobes**
Module for checking updates on a Fedora distribution, similar to the arch_updates module.

### CODE REVIEW

File: py3status/modules/dnf.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        output, error = subprocess.Popen(
+            ['dnf', 'check-update'],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE).communicate()
+
+        updates = len(self._reg_ex_pkg.findall(output))
+
+        if updates == 0:
+            color = self.color_good or i3s_config['color_good']
+            results = self.format.format(updates='')
```
Comment (pydsigner) [inline py3status/modules/dnf.py]: Why `updates=''` instead of passing in the `0`?

File: py3status/modules/dnf.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+to be installed.
+Additionally check if any update security notices.
+
+Configuration parameters:
+    - cache_timeout : how often we refresh this module in seconds
+      default is 600
+    - format : display format
+      default is 'DNF: {updates}'
+    - color_good : color when no upgrades needed
+      default is None
```
Comment (pydsigner) [inline py3status/modules/dnf.py]: I'd suggest changing "default is None" to "defaults to global color_good" and so on for the other two colors.

**Comment by ultrabug** [comment]
Thanks for your contribution @tobes !

To add to what @pydsigner said, I'd like to suggest to rename this module to something more understandable for non-fedora users too, like `fedora_dnf` maybe ? OR to stay in line with the arch module, why not `fedora_updates` ?

**Comment by pydsigner** [comment]
I'll agree with fedora_updates. But fedora_dnf for people who don't know what dnf is seems like putting mediaplayer_deadbeef or _spotify for people who don't know what those are. A user looking through the module listings will get more information anyways.

File: py3status/modules/dnf.py (hunk 1 above)
Comment (tobes) [inline py3status/modules/dnf.py]: I've changed this so now 0 if no updates and ? when unknown

File: py3status/modules/dnf.py (hunk 2 above)
Comment (tobes) [inline py3status/modules/dnf.py]: Done.  I copied that from another module but yes this is better

**Comment by tobes** [comment]
Renamed to `fedora_updates` and other suggested changes.  Let me know if it needs and other work.  or if you want a fresh pull request as a single commit?

File: py3status/modules/fedora_updates.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+"""
+Displays the number of package updates pending for a Fedora Linux installation.
+This will display a count of how many `dnf` updates are waiting
+to be installed.
+Additionally check if any update security notices.
+
+Configuration parameters:
+    - cache_timeout : how often we refresh this module in seconds
```
Comment (ultrabug) [inline py3status/modules/fedora_updates.py]: please order the parameters alphabetically

File: py3status/modules/fedora_updates.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+from time import time
+import subprocess
+import re
+
+
+class Py3status:
+    # available configuration parameters
+    cache_timeout = 600
+    format = 'DNF: {updates}'
+    color_good = None
```
Comment (ultrabug) [inline py3status/modules/fedora_updates.py]: please order the parameters alphabetically

File: py3status/modules/fedora_updates.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    color_bad = None
+    color_degraded = None
+
+    _reg_ex_sec = re.compile('\d+(?=\s+Security)')
```
Comment (ultrabug) [inline py3status/modules/fedora_updates.py]: those non configuration parameters may be better in a proper __init__(self) function please

**Comment by ultrabug** [comment]
Last comments in ! thx :)

**Comment by tobes** [comment]
ok, I've made the changes

**Pull request closed without merging**